lib/analytics.js

```diff
@@ -791,12 +816,53 @@ Analytics.prototype._invoke = function(method, facade) {
                 result = new Facade(result);
               }
 
-              metrics.increment('analytics_js.integration.invoke', {
-                method: method,
-                integration_name: integration.name
-              });
-
-              integration.invoke.call(integration, method, result);
+              // apply destination middlewares
+              // Apply any integration middlewares that exist, then invoke the integration with the result.
+              if (self._destinationMiddlewares[integration.name]) {
+                self._destinationMiddlewares[integration.name].applyMiddlewares(
+                  facadeCopy,
+                  integration.name,
+                  function(result) {
+                    // A nullified payload should not be sent to an integration.
+                    if (result === null) {
+                      self.log(
+                        'Payload to destination "%s" was null and dropped by a middleware.',
+                        name
+                      );
+                      return;
+                    }
+
+                    // Check if the payload is still a Facade. If not, convert it to one.
+                    if (!(result instanceof Facade)) {
+                      result = new Facade(result);
+                    }
+
+                    self.emit('invoke', result);
```

Comment:
@aultimus thinking more about this, we need to put this back outside in the place that it was before. Right now we are doing `self.emit('invoke', result);` multiple times. Let's say the customer has enabled 10 integrations, we will emit the `invoke` 10 times ( and subsequently segment's own integration will record 10 events ). Right now the `invoke` event seems to be integration agnostic (unlike the `integration.invoke.call(integration, method, result);` call), therefore it should be triggered regardless of the destination middleware.